models/users.js

```diff
@@ -178,26 +179,38 @@ const fetchPaginatedUsers = async (query) => {
   }
 };
 
-const fetchAllUsers = async () => {
+const fetchUsers = async (usernames = []) => {
   try {
     const dbQuery = userModel;
+    const filterdUsersWithDetails = [];
 
-    const snapshot = await dbQuery.get();
+    const groups = [];
+    for (let i = 0; i < usernames.length; i += BATCH_SIZE_IN_CLAUSE) {
+      groups.push(usernames.slice(i, i + BATCH_SIZE_IN_CLAUSE));
+    }
 
-    const allUsers = [];
+    // For each group, write a separate query
+    const promises = groups.map((group) => {
+      return dbQuery.where("github_id", "in", group).get();
+    });
 
-    snapshot.forEach((doc) => {
-      allUsers.push({
-        id: doc.id,
-        ...doc.data(),
-        phone: undefined,
-        email: undefined,
-        tokens: undefined,
-        chaincode: undefined,
+    const snapshots = await Promise.all(promises);
+
+    snapshots.forEach((snapshot) => {
+      snapshot.forEach((doc) => {
+        filterdUsersWithDetails.push({
+          id: doc.id,
+          ...doc.data(),
+          phone: undefined,
+          email: undefined,
+          tokens: undefined,
+          chaincode: undefined,
+        });
       });
     });
+
     return {
-      allUsers,
+      filterdUsersWithDetails,
```

Comment:
We should rename this to something  like `users` instead of `filteredUsersWithDetails`, or to something else

Reply:
Good point Prakash. Unfortunately, I jumped the gun and merged 😢

Reply:
Will get someone to fix it here: https://github.com/Real-Dev-Squad/website-backend/issues/1026